### PR TITLE
Feature: Service Set As a File

### PR DESCRIPTION
**Pull request by pareankit29**
# Odin version(x.y.z)

## Features
Accept service set definition as a file and deploy the services without creating a service set.

### CODE REVIEW

File: pkg/request/request.go [hunk 1]
```diff
@@ -103,6 +103,35 @@ func (r *Response) Process(exitOnError bool) {
...
+// Process Request Response and Return Status Code
+
+func (r *Response) GetStatusCode(exitOnError bool) int {
+
+	if r.Error != nil {
+		logger.Error(r.Error.Error())
+		HandleExit(1, exitOnError)
+	} else {
+		if MatchStatusCode(r.StatusCode, 200) {
+			logger.Debug(r.Status)
```
Comment (akshaypatidar1999) [inline pkg/request/request.go:115]: Is this needed?

File: internal/command/commands/serviceset.go [hunk 2]
```diff
@@ -26,6 +26,7 @@ func (s *ServiceSet) Run(args []string) int {
 	flagSet := flag.NewFlagSet("flagSet", flag.ContinueOnError)
 	// create flags
 	filePath := flagSet.String("file", "service-set.json", "file to read service-set config")
+	setFilePath := flagSet.String("setFile", "", "file to read temporary service-set config")
```
Comment (akshaypatidar1999) [inline internal/command/commands/serviceset.go:29]: See if we can use same flag

File: internal/command/commands/serviceset.go [hunk 3]
```diff
@@ -140,13 +143,99 @@ func (s *ServiceSet) Run(args []string) int {
...
+			return 1
+		}
+		if isFilePresent && isServiceSetNamePresent {
+			s.Logger.Error("--name should not be provided when --setFile is provided.")
+			return 1
+		} else if !isFilePresent && !isServiceSetNamePresent {
+			s.Logger.Error("Please provide either --name or --setFile.")
+			return 1
+		}
+		emptyFileCreatedParameters := emptyParameters(map[string]string{"--env": *envName, "--setFile": *setFilePath})
```
Comment (akshaypatidar1999) [inline internal/command/commands/serviceset.go:161]: We can remove this?
DIrectly add condition if `setFilePath` is empty or not

File: internal/command/commands/serviceset.go [hunk 4]
```diff
@@ -140,13 +143,99 @@ func (s *ServiceSet) Run(args []string) int {
...
+			serviceDataMap := parsedConfig.(map[string]interface{})
+
+			serviceSetClient.CreateTempServiceSet(parsedConfig)
+			s.Logger.Success(fmt.Sprintf("Temporary ServiceSet: %s initialised Successfully.", serviceDataMap["name"].(string)))
+
+			var forceDeployServices []serviceset.ListEnvService
+			if !*force {
+				//get list of env services
+				s.Logger.Debug(fmt.Sprintf("Env Services of service-set %s and env %s", *serviceSetName, *envName))
+				serviceSetList, err := serviceSetClient.ListEnvServices(fmt.Sprintf("%v", serviceDataMap["name"]), *envName, "conflictedVersion", true)
```
Comment (akshaypatidar1999) [inline internal/command/commands/serviceset.go:197]: See if we can create a method from this as similar code is present in deploy with name

File: internal/command/commands/serviceset.go [hunk 5]
```diff
@@ -97,8 +98,9 @@ func (s *ServiceSet) Run(args []string) int {
 	if s.Describe {
 		emptyParameters := emptyParameters(map[string]string{"--name": *serviceSetName})
 		if len(emptyParameters) == 0 {
+			isFile := serviceSetClient.IdentifyServiceSetType(*serviceSetName)
```
Comment (akshaypatidar1999) [inline internal/command/commands/serviceset.go:101]: Do we need to describe service set?

**Review by akshaypatidar1999 — APPROVED** [review]

**Review by sarveshdream11 — APPROVED** [review]

**Pull request merged**